Lib/asyncore.py

```diff
@@ -181,11 +178,12 @@ def poll2(timeout=0.0, map=None):
                 pollster.register(fd, flags)
 
         r = pollster.poll(timeout)
-        for fd, flags in r:
-            obj = map.get(fd)
-            if obj is None:
-                continue
-            readwrite(obj, flags)
+        ready = [(map[fd], flags) for fd, flags in r]
+
+        for obj, flags in ready:
+            if not obj.closing:
```

Comment:
But if you store all the objects on ready you can't call just call all of them because an object previous in the list might have closed a posterior one.